code/modules/reagents/chemistry/recipes.dm

```diff
@@ -339,56 +347,177 @@
+			movey.throw_at(throw_target, moving_power, 1)
 
 
-//Creates a ring of fire in a set range around the beaker location
-/datum/chemical_reaction/proc/explode_fire(datum/reagents/holder, datum/equilibrium/equilibrium, range)
-	explosion(holder.my_atom, 0, 0, 0, 0, flame_range = 3)
+////////BEGIN FIRE BASED EXPLOSIONS
+
+//Calls the default explosion subsystem handiler to explode with fire (random firespots and noise)
+/datum/chemical_reaction/proc/explode_fire(datum/reagents/holder, datum/equilibrium/equilibrium, range = 3)
+	explosion(holder.my_atom, 0, 0, 0, 0, flame_range = range)
 	holder.my_atom.audible_message("The [holder.my_atom] suddenly errupts in flames!")
 
-//Clears the beaker of the reagents only
-/datum/chemical_reaction/proc/clear_reactants(datum/reagents/holder, volume = null)
+//Creates a ring of fire in a set range around the beaker location
+/datum/chemical_reaction/proc/explode_fire_vortex(datum/reagents/holder, datum/equilibrium/equilibrium, x_offset = 1, y_offset = 1, reverse = FALSE, id = "f_vortex", )
+	var/increment = reverse ? -1 : 1
+	if(!equilibrium.data)
+		equilibrium.data = list("[id]_x" = x_offset, "[id]_y" = y_offset, "[id]_tar" = "[id]_y")//tar is the current movement direction the cyclone is moving in
+	if(equilibrium.data["[id]_tar"] == "[id]_x")
+		if(equilibrium.data["[id]_x"] >= x_offset)
+			equilibrium.data["[id]_tar"] = "[id]_y"
+			equilibrium.data["[id]_y"] += increment
+		else if(equilibrium.data["[id]_x"] <= -x_offset)
+			equilibrium.data["[id]_tar"] = "[id]_y"
+			equilibrium.data["[id]_y"] -= increment
+		else
+			if(equilibrium.data["[id]_y"] < 0)
+				equilibrium.data["[id]_x"] += increment
+			else if(equilibrium.data["[id]_y"] > 0)
+				equilibrium.data["[id]_x"] -= increment
+
+	else if (equilibrium.data["[id]_tar"] == "[id]_y")
+		if(equilibrium.data["[id]_y"] >= y_offset)
+			equilibrium.data["[id]_tar"] = "[id]_x"
+			equilibrium.data["[id]_x"] -= increment
+		else if(equilibrium.data["[id]_y"] <= -y_offset)
+			equilibrium.data["[id]_tar"] = "[id]_x"
+			equilibrium.data["[id]_x"] += increment
+		else
+			if(equilibrium.data["[id]_x"] < 0)
+				equilibrium.data["[id]_y"] -= increment
+			else if(equilibrium.data["[id]_x"] > 0)
+				equilibrium.data["[id]_y"] += increment
+	var/turf/holder_turf = get_turf(holder.my_atom)
+	var/turf/target = locate(holder_turf.x + equilibrium.data["[id]_x"], holder_turf.y + equilibrium.data["[id]_y"], holder_turf.z)
+	new /obj/effect/hotspot(target)
+	debug_world("X: [equilibrium.data["[id]_x"]], Y: [equilibrium.data["[id]_x"]]")
+
+/*
+ * Creates a square of fire in a fire_range radius,
+ * fire_range = 0 will be on the exact spot of the holder,
+ * fire_range = 1 or more will be additional tiles around the holder. Every tile will be heated this way.
+ * How clf3 works, you know!
+ */
+/datum/chemical_reaction/proc/explode_fire_square(datum/reagents/holder, datum/equilibrium/equilibrium, fire_range = 1)
+	var/turf/T = get_turf(holder.my_atom)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Reply:
Changed to location